src/nwp_consumer/internal/inputs/noaa/test_ncar.py

```diff
@@ -0,0 +1,43 @@
+import datetime as dt
+import pathlib
+import unittest
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ._models import NOAAFileInfo
+
+from .ncar import Client, _parseNCARFilename
+
+testClient = Client(model="global", param_group="full")
+
+
+class TestClient(unittest.TestCase):
+    def test_mapTemp(self) -> None:
+        # Test with global file
+        testFilePath: pathlib.Path = (
+            pathlib.Path(__file__).parent / "gfs.0p25.2023121906.f001.grib2"
+        )
+        out = testClient.mapTemp(p=testFilePath)
+        # Check latitude and longitude are injected
+        self.assertTrue("latitude" in out.coords)
+        self.assertTrue("longitude" in out.coords)
+        # Check that the dimensions are correctly ordered and renamed
+        self.assertEqual(
+            out[next(iter(out.data_vars.keys()))].dims,
+            ("init_time", "step", "latitude", "longitude"),
```

Comment:
I tend to check that these also have the correct sizes, just in case there's some unknown problem where some bits are getting missed from the grib conversion process. Not a blocker though.

Reply:
Done!